ports/stm32h7/port.mk

```diff
@@ -0,0 +1,53 @@
+UF2_FAMILY_ID = 0x6db66082 
+CROSS_COMPILE = arm-none-eabi-
+
+ST_HAL_DRIVER = lib/st/stm32h7xx_hal_driver
+ST_CMSIS = lib/st/cmsis_device_h7
+CMSIS_5 = lib/CMSIS_5
+
+# Compiler flags
+CFLAGS += \
+  -flto \
```

Comment:
I think LTO is what "broke" code for me (probably gcc-version dependant issue if it worked for you...), do we really need to optimize tinyuf2 code if we aren't using the built-in flash?
Perhaps individual boards using H7xx could opt-in this config, instead of "forcing" its use having it at the MCU level

Reply:
That's a good idea, I copied the F4-mkfiles for this port, so there maybe a few more gotchas.
what version of GCC are you compiling with?

Reply:
10.3.1 (from ubuntu on WSL2 if that matters)
```
elpekenin@elPeCenin:~$ arm-none-eabi-gcc --version
arm-none-eabi-gcc (15:10.3-2021.07-4) 10.3.1 20210621 (release)
```

---

I cant get the flashed `.uf2` to run, i just get back to bootloader every single time, do you happen to know what im doing wrong here? <https://github.com/elpekenin/qmk_firmware/blob/579e150c72899e78f06e42aefc27d1b71dcaf7ba/keyboards/handwired/onekey/weact_stm32h750vbt6/ld/STM32H750xB_qspi_tinyuf2.ld>. Based this off the `.ld` on ChibiOS and did also take a look on a `.ld` on your GitHub

Reply:
> arm-none-eabi-gcc (15:10.3-2021.07-4) 10.3.1 20210621 (release)

vs 

> Using built-in specs.
COLLECT_GCC=arm-none-eabi-gcc
COLLECT_LTO_WRAPPER=~/Workspace/toolchains/gcc-arm-none-eabi-10.3-2021.10/bin/../lib/gcc/arm-none-eabi/10.3.1/lto-wrapper
Target: arm-none-eabi
Configured with: <removed>
Thread model: single
Supported LTO compression algorithms: zlib
gcc version 10.3.1 20210824 (release) (GNU Arm Embedded Toolchain 10.3-2021.10) 

Definitely seems like something LTO -related is different between our toolchains.

---

 I'm not sure how qmk is laid out in memory, but in your ld, can you try creating 2 memory regions? 
Something like
```
    flash0 : org = 0x90000000, len = 0x300
    flash1 : org = 0x90000300, len = 8m-0x300
```
to explicitly ensure that the vector table comes first

Reply:
The build should output a .map file to show you the final memory layout. The vector table should have its out input section to ensure it is placed correctly.

Reply:
I feel like we should move this to Discord or at least an issue, to stop cluttering the PR (sorry), please answer this message in any way and @ me into it. Unless you think it's better staying here, that is.

The map file looks correct(ish)? Idk, im as noob as you can be with ld stuff....
<https://gist.github.com/elpekenin/95eab3c7831187b86f29b983050e5a4e> (line 2551)